examples/validate.py

```diff
@@ -91,14 +91,22 @@ def validateExamples(examples, schemas, maxExamples, shuffle):
 
   numChecked = 0
   latestReportTime = time.perf_counter()
-  
+  # Keys in schemas are $TYPE-$VERSION.
+  # All files are at ./$TYPE/$VERSION.json.
+  # Links from file to other file are always ../$OTHER_TYPE/$VERSION.json.
+  schema_store = {
+    "../" + key.replace("-", "/") + ".json": schemas[key] for key in schemas
+  }
+  resolver = RefResolver(None, referrer=None, store=schema_store)
+
```

Comment:
This feels a bit icky. If we create a RefResolver with a base URI (i.e. first positional argument) that matches the path to the original schema it'll load the EiffelMetaProperty by itself. I replaced line 108, below, with
```python
        validator = Draft4Validator(
          schemas[schemaKey],
          resolver=RefResolver(
            "file://" + os.path.join(
              os.getcwd(),
              "schemas/EiffelCompositionDefinedEvent/foo",
            ),
            None,
          ),
        )
```
and it worked fine. The "foo" part is obviously bogus and should be the actual path to the schema file.

We should probably replace the whole `schemas` dict with a dict that maps from (type, version) tuples to validator instances. I think that would improve the code in general.

Reply:
It would make total sense to keep the validators in a collection instead of the current schema dict.
The validators would have to share resolver (with all available schemas) since we don't know what other schema files, the schema we are using to validate, references. Unless we want to enumerate the "sub-schema" used (only meta in this case but in the future potentially more) and populate that list when we add a new "sub-schema".

Reply:
Done